app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryViewModel.kt

```diff
@@ -42,4 +42,16 @@ class PromotedStoryViewModel(
       promotedStory.storyId
     )
   }
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) {
+      return true
+    }
+    if (other == null || other.javaClass != javaClass) {
+      return false
+    }
+    val otherResult = other as PromotedStoryViewModel
+    return otherResult.internalProfileId == this.internalProfileId &&
```

Comment:
Suggest also checking the total story count & entity type fields.

Reply:
Done